### PR TITLE
Icu 764- Fixes username mentions at the end of a sentence

### PR DESCRIPTION
**Pull request by stephenkiers**
#### Summary
Posting @user.name followed by "." (like "@user.name.") does not appear mention jewel

#### Ticket Link
https://mattermost.atlassian.net/browse/ICU-764

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: app/notification.go [hunk 1]
```diff
@@ -832,6 +832,9 @@ func GetExplicitMentions(message string, keywords map[string][]string) *Explicit
 				continue
 			}
 
+			// remove trailing '.', as that is the end of a sentence
+			word = strings.TrimSuffix(word, ".")
+
```
Comment (stephenkiers) [inline app/notification.go]: This simply removes the period at the end of the split. So 
- `@username.` becomes `@username` and gets properly evaluated in the first section (pre-886). It used to get caught in the second block when it was split on the period.
- `@user.name.` becomes `@user.name` and gets properly caught in the first section (pre-886). In the old code it isn't caught in the first section, and it becomes `@user` and `name` after 886, thus never caught.

**Comment by stephenkiers** [comment]
Added new tests and refactored code.

Now, with users [1]`@username`, [2]`@user.name` and [3]`@user.name.`, [4]`@diff.name` the follow would tag

- `hello @username` tags 1
- `hello @username.` tags 1
- `hello @username.name` tags 2
- `hello @username.name.` tags 3
- `hello @diff.name` tags 4
- `hello @diff.name.` tags 4

File: app/notification.go [hunk 2]
```diff
@@ -820,43 +820,54 @@ func GetExplicitMentions(message string, keywords map[string][]string) *Explicit
 		}
 	}
 
+	checkForMention := func(word string) bool {
+		isMention := false
+
+		fmt.Printf("New Word: %v\n", word)
```
Comment (jwilander) [inline app/notification.go]: Forgot to remove this debug statement?

Comment (stephenkiers) [reply app/notification.go]: Thanks. Will remove.

**Comment by stephenkiers** [comment]
Why was jenkins passsing with debug statement and failing now?

**Pull request merged**